sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_patch.py

```diff
@@ -0,0 +1,141 @@
+    """Configuration for WebPubSubServiceClient.
+
+    Note that all parameters used to create this instance are saved as instance
+    attributes.
+
+    :param endpoint: HTTP or HTTPS endpoint for the Web PubSub service instance.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: Union[~azure.core.credentials_async.AsyncTokenCredential, ~azure.core.credentials.AzureKeyCredential]
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        credential: Union["AsyncTokenCredential", "AzureKeyCredential"],
+        **kwargs: Any
+    ) -> None:
+        if endpoint is None:
+            raise ValueError("Parameter 'endpoint' must not be None.")
+        if credential is None:
+            raise ValueError("Parameter 'credential' must not be None.")
+        super(WebPubSubServiceClientConfiguration, self).__init__(**kwargs)
+
+        self.endpoint = endpoint
+        self.credential = credential
+        self.credential_scopes = kwargs.pop('credential_scopes', ['https://webpubsub.azure.com/.default'])
+        kwargs.setdefault('sdk_moniker', 'messaging-webpubsubservice/{}'.format(VERSION))
+        self._configure(**kwargs)
+
+    def _configure(
+        self,
+        **kwargs: Any
+    ) -> None:
+        self.user_agent_policy = kwargs.get('user_agent_policy') or policies.UserAgentPolicy(**kwargs)
+        self.headers_policy = kwargs.get('headers_policy') or policies.HeadersPolicy(**kwargs)
+        self.proxy_policy = kwargs.get('proxy_policy') or policies.ProxyPolicy(**kwargs)
+        self.logging_policy = kwargs.get('logging_policy') or policies.NetworkTraceLoggingPolicy(**kwargs)
+        self.http_logging_policy = kwargs.get('http_logging_policy') or policies.HttpLoggingPolicy(**kwargs)
+        self.retry_policy = kwargs.get('retry_policy') or policies.AsyncRetryPolicy(**kwargs)
+        self.custom_hook_policy = kwargs.get('custom_hook_policy') or ApiManagementProxy(**kwargs)
+        self.redirect_policy = kwargs.get('redirect_policy') or policies.AsyncRedirectPolicy(**kwargs)
+        self.authentication_policy = kwargs.get('authentication_policy')
+        if self.credential and not self.authentication_policy:
+            if isinstance(self.credential, AzureKeyCredential):
+                self.authentication_policy = JwtCredentialPolicy(self.credential, kwargs.get('user'))
+            else:
+                self.authentication_policy = policies.AsyncBearerTokenCredentialPolicy(self.credential, *self.credential_scopes, **kwargs)
+
+
+class WebPubSubServiceClient(GeneratedWebPubSubServiceClient):
+    """WebPubSubServiceClient.
+
+    :param endpoint: HTTP or HTTPS endpoint for the Web PubSub service instance.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: ~azure.core.credentials_async.AsyncTokenCredential
```

Comment:
can also be `~azure.core.credentials.AzureKeyCredential`

Reply:
same with up comment.